package.json

```diff
@@ -115,69 +115,87 @@
                 "title": "SFDX: Scan Current File for Performance Issues with ApexGuru"
             }
         ],
-        "configuration": {
-            "title": "Salesforce Code Analyzer",
-            "properties": {
-                "codeAnalyzer.pMD.customConfigFile": {
+        "configuration": [{
+			"title": "General",
+			"properties": {
+				"codeAnalyzerV5.enabled": {
+					"type": "boolean",
+					"default": false,
+					"description": "Use Code Analyzer v5 (Beta) instead of Code Analyzer v4."
+				},
+                "codeAnalyzer.analyzeOnSave.enabled": {
+                    "type": "boolean",
+                    "default": false,
+                    "description": "Scan files on save automatically."
+                },
+                "codeAnalyzer.analyzeOnOpen.enabled": {
+                    "type": "boolean",
+                    "default": false,
+                    "description": "Scan files on open automatically."
+                },
+				"codeAnalyzer.apexGuru.enabled": {
+                    "type": "boolean",
+                    "default": false,
+                    "description": "Discover critical problems and performance issues in your Apex code with ApexGuru, which analyzes your Apex files for you. This feature is in a closed pilot; contact your account representative to learn more."
+                }
+			}
+		}, {
+			"title": "Code Analyzer v5 (Beta)",
+			"properties": {
+				"codeAnalyzer.ruleSelectors": {
```

Comment:
This one uses "Code Analyzer" instead of "Code Analyzer v5" because it's going to be a permanent setting and I feel weird about "v5" being part of the name even after we deprecate v4.